Sources/MapboxMaps/Location/Pucks/Puck.swift

```diff
@@ -8,6 +8,9 @@ internal protocol Puck {
     /// Property that stores the current `PuckStyle` of the puck
     var puckStyle: PuckStyle { get set }
 
+    /// Property that stores the current `PuckBearingSource` of the puck
```

Comment:
We don't need both of these right? I think we should just use the `puckBearingSource` property.. We can get rid of the `updateBearingSource` method.